### PR TITLE
enabling granular error reporting for expression matrix validation (SCP-2660)

### PR DESCRIPTION
**Pull request by devonbush**
This allows reporting of the specific errors for each validation failure.

### CODE REVIEW

File: ingest/expression_files/expression_files.py [hunk 1]
```diff
@@ -97,7 +97,14 @@ def has_unique_cells(cell_names: List, study_id, client):
             for cell_values in query_results
             for values in cell_values.get("values")
         ]
-        return not any(name in existing_cells for name in cell_names)
+        dupes = set(existing_cells) & set(cell_names)
+        if len(dupes) != 0:
+            error_string = f'Expression file contains {len(dupes)} cells that also exist in another file. '
```
Comment (eweitz) [inline ingest/expression_files/expression_files.py:102]:
```suggestion
            error_string = f'Expression file contains {len(dupes)} cells that also exist in another expression file. '
```

**Comment by eweitz** [comment]
I added SCP-2660 to the title for Jira traceability.

**Comment by devonbush** [comment]
> I added [SCP-2660](https://broadworkbench.atlassian.net/browse/SCP-2660) to the title for Jira traceability.

thanks!

File: ingest/expression_files/dense_ingestor.py [hunk 2]
```diff
@@ -47,29 +47,46 @@ def execute_ingest(self):
...
-    def is_valid(header, row, query_params):
-        return all(
-            [
-                DenseIngestor.has_unique_header(header),
-                DenseIngestor.has_gene_keyword(header, row),
-                DenseIngestor.header_has_valid_values(header),
-                GeneExpression.has_unique_cells(header, *query_params),
-            ]
-        )
+    def check_valid(header, first_row, query_params):
```
Comment (knapii-developments) [inline ingest/expression_files/dense_ingestor.py:63]: I wouldn't have thought of structuring this method with try catch blocks. Cool!

**Pull request merged**